python/paddle/fluid/tests/unittests/test_nan_to_num_op.py

```diff
@@ -0,0 +1,203 @@
+            else paddle.CPUPlace()
+        )
+
+    def test_static(self):
+        x_np = np.array([[1, np.nan, -2], [np.inf, 0, -np.inf]]).astype(
+            np.float32
+        )
+        out1_np = np_nan_to_num(x_np)
+        out2_np = np_nan_to_num(x_np, 1.0)
+        out3_np = np_nan_to_num(x_np, 1.0, 9.0)
+        out4_np = np_nan_to_num(x_np, 1.0, 9.0, -12.0)
+        paddle.enable_static()
+        with paddle.static.program_guard(paddle.static.Program()):
+            x = paddle.fluid.data('X', x_np.shape)
+            out1 = paddle.nan_to_num(x)
+            out2 = paddle.nan_to_num(x, 1.0)
+            out3 = paddle.nan_to_num(x, 1.0, 9.0)
+            out4 = paddle.nan_to_num(x, 1.0, 9.0, -12.0)
+            exe = paddle.static.Executor(self.place)
+            res = exe.run(feed={'X': x_np}, fetch_list=[out1, out2, out3, out4])
+
+        self.assertTrue(np.allclose(out1_np, res[0]))
+        self.assertTrue(np.allclose(out2_np, res[1]))
+        self.assertTrue(np.allclose(out3_np, res[2]))
+        self.assertTrue(np.allclose(out4_np, res[3]))
+
+    def test_dygraph(self):
+
+        paddle.disable_static(place=self.place)
+
+        with paddle.fluid.dygraph.guard():
+            # NOTE(tiancaishaonvjituizi): float64 input fails the test
+            x_np = np.array([[1, np.nan, -2], [np.inf, 0, -np.inf]]).astype(
+                np.float32
+                # np.float64
+            )
+            x_tensor = paddle.to_tensor(x_np, stop_gradient=False)
+
+            out_tensor = paddle.nan_to_num(x_tensor)
+            out_np = np_nan_to_num(x_np)
+            self.assertTrue(np.allclose(out_tensor.numpy(), out_np))
+
+            out_tensor = paddle.nan_to_num(x_tensor, 1.0, None, None)
+            out_np = np_nan_to_num(x_np, 1, None, None)
+            self.assertTrue(np.allclose(out_tensor.numpy(), out_np))
+
+            out_tensor = paddle.nan_to_num(x_tensor, 1.0, 2.0, None)
+            out_np = np_nan_to_num(x_np, 1, 2, None)
+            self.assertTrue(np.allclose(out_tensor.numpy(), out_np))
+
+            out_tensor = paddle.nan_to_num(x_tensor, 1.0, None, -10.0)
+            out_np = np_nan_to_num(x_np, 1, None, -10)
+            self.assertTrue(np.allclose(out_tensor.numpy(), out_np))
+
+            out_tensor = paddle.nan_to_num(x_tensor, 1.0, 100.0, -10.0)
+            out_np = np_nan_to_num(x_np, 1, 100, -10)
+            self.assertTrue(np.allclose(out_tensor.numpy(), out_np))
+
+        paddle.enable_static()
+
+    def test_check_grad(self):
+        paddle.disable_static(place=self.place)
+        x_np = np.array([[1, np.nan, -2], [np.inf, 0, -np.inf]]).astype(
+            np.float32
+        )
+        x_tensor = paddle.to_tensor(x_np, stop_gradient=False)
+
+        y = paddle.nan_to_num(x_tensor)
+        dx = paddle.grad(y, x_tensor)[0].numpy()
+
+        np_grad = np_nan_to_num_grad(x_np, np.ones_like(x_np))
+        self.assertTrue(np.allclose(np_grad, dx))
```

Comment:
```
2022-10-25 11:07:01 0. It is recommended to use 'np.testing.assert_allclose' and 'np.testing.array_equal' instead of 'self.assertTrue(np.allclose(...))' and 'self.assertTrue(np.array_equal(...))'.
2022-10-25 11:07:01 Please modify the code below. If anything is unclear, please read the specification [ https://github.com/PaddlePaddle/community/blob/master/rfcs/CodeStyle/20220805_code_style_improvement_for_unittest.md#background ]. If it is a mismatch, please request qili93 (Recommend) or luotao1 review and approve.
```
<img width="718" alt="image" src="https://user-images.githubusercontent.com/6836917/197698145-6c705dff-3379-4f5c-b0d0-2832a1f3692a.png">

单测报错信息进行了升级，可以在下一个PR里根据提示使用`np.testing.assert_allclose`代替`self.assertTrue(np.allclose`